src/libs/OptionsListUtils.ts

```diff
@@ -1542,6 +1549,81 @@ function orderOptions(options: ReportUtils.OptionData[], searchValue: string | u
     );
 }
 
+function canCreateOptimisticPersonalDetailOption({
+    searchValue,
+    recentReportOptions,
+    personalDetailsOptions,
+    currentUserOption,
+    selectedOptions,
+    excludeUnknownUsers,
+    betas,
+    optionsToExclude,
+    parsedPhoneNumber,
+}: {
+    searchValue: string;
+    recentReportOptions: ReportUtils.OptionData[];
+    personalDetailsOptions: ReportUtils.OptionData[];
+    currentUserOption?: ReportUtils.OptionData | null;
+    selectedOptions: Array<Partial<ReportUtils.OptionData>>;
+    excludeUnknownUsers: boolean;
+    betas: OnyxEntry<Beta[]>;
+    optionsToExclude: string[];
+    parsedPhoneNumber: ParsedPhoneNumber;
+}) {
+    const noOptions = recentReportOptions.length + personalDetailsOptions.length === 0 && !currentUserOption;
+    const noOptionsMatchExactly = !personalDetailsOptions
```

Comment:
Just noting that this looks like a potentially very expensive call - we're looping through all personalDetails (there may be many), doing a regex search + replace for each.

Reply:
yeah that is true, however not sure how to tackle that in a different way. It worked this way until now, I just extracted it into a separate function, so I think there is no need to prematurely optimize it, but for sure it might become a potential bottleneck once it hits the big amount of data

Reply:
Thanks yeah I see you just extracted it to a separate function, we can leave this. This was partially a note-to-self as I was looking through the code